### PR TITLE
Add regexp_match scalar function (#15117)

### LINKED ISSUE #15117: Add `regexp_match` scalar
### Problem Statement

CrateDB contains a `regexp_matches` table function, but for some use cases a regular scalar would be the better fit. One use-case is using them in generated columns, where table functions cannot be used.

From https://github.com/crate/crate/issues/15077:

```sql
CREATE TABLE hernan.test  (
	a TEXT,
	b ARRAY(TEXT) GENERATED ALWAYS AS regexp_match(a, '(a(.+)z)')
	);
	
INSERT INTO hernan.test (a) 
SELECT 'alcatraz';
```

### Possible Solutions

PostgreSQL provides a `regexp_match` scalar. See https://www.postgresql.org/docs/16/functions-string.html

`regexp_match ( string text, pattern text [, flags text ] ) → text[]`

Returns substrings within the first match of the POSIX regular expression pattern to the string; see [Section 9.7.3](https://www.postgresql.org/docs/current/functions-matching.html#FUNCTIONS-POSIX-REGEXP).

```
regexp_match('foobarbequebaz', '(bar)(beque)') → {bar,beque}
```

See: https://github.com/crate/crate/blob/ff1e7f7b882f1048be6d7806023674703cc2720a/server/src/main/java/io/crate/expression/tablefunctions/MatchesFunction.java#L54 for the current `regexp_matches` implementation. The main difference is the support for the global (`g`) flag, that can result in more than one result.

### PR DESCRIPTION
**Pull request by gohri81-tech**
## Summary

Implements the PostgreSQL-compatible `regexp_match` scalar function requested in #15117.

### Changes

* Add `regexp_match(string, pattern)` support
* Add optional regex flags argument
* Return captured groups when the pattern contains capturing groups
* Return the complete match when there are no capturing groups
* Return `NULL` when there is no match
* Add unit tests covering matches, capturing groups, flags, and NULL values

Closes #15117

### CODE REVIEW

**Comment by gohri81-tech** [comment]
I have now signed the CrateDB Contributor License Agreement. Could a maintainer please remove the `needs CLA` label so the CLA check re-runs? Thanks!

**Comment by matriv** [comment]
ok to test

File: server/src/test/java/io/crate/expression/scalar/regex/RegexpMatchFunctionTest.java [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    }
+
+    @Test
+    public void test_match_with_unmatched_group() {
+        assertEvaluate("regexp_match('b', '(a)?b')", new String[]{null}, Literal.of("b"));
+    }
+
+    @Test
+    public void test_flags() {
+        assertEvaluate("regexp_match(name, '(BAR)', 'i')", new String[]{"BAR"}, Literal.of("foobarbequebaz"));
```
Comment (matriv) [inline server/src/test/java/io/crate/expression/scalar/regex/RegexpMatchFunctionTest.java]: postgres returns the match from the string, not as passed in the pattern argument:
```
matriv=# select regexp_match('foobARbequebaz', '(BAR)', 'i');
 regexp_match
--------------
 {bAR}
(1 row)

matriv=# select regexp_match('foobarbequebaz', '(BAR)', 'i');
 regexp_match
--------------
 {bar}
(1 row)
```

**Review by matriv — COMMENTED** [review]
Thank you for the contribution, please check the comment I left regarding mismatched behavior vs postgres. Additionally, please add docs and release note entry for the new function.

**Review by matriv — APPROVED** [review]
thx!
